### PR TITLE
3-develop-static-page_crawlability

### PR DESCRIPTION
**Pull request by NathanBurkett**
Ref #3

### Adds:
- Deterministic access to sitemap.xml and robots.txt

### CODE REVIEW

File: src/app/Http/Controllers/RobotsTxtController.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+class RobotsTxtController extends Controller
+{
+    /**
+     * @param IndexingAccessService $indexingService
+     * @param ResponseFactory       $response
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index(IndexingAccessService $indexingService, ResponseFactory $response)
```
Comment (NathanBurkett) [inline src/app/Http/Controllers/RobotsTxtController.php]: ResponseFactory already an accessible member on the parent as $responseFactory

File: src/app/Http/Controllers/SiteMapController.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+class SiteMapController extends Controller
+{
+    /**
+     * @param IndexingAccessService $indexingService
+     * @param ResponseFactory       $response
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index(IndexingAccessService $indexingService, ResponseFactory $response)
```
Comment (NathanBurkett) [inline src/app/Http/Controllers/SiteMapController.php]: Same as above

File: src/app/Http/Controllers/RobotsTxtController.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        if ($indexingService->accessAllowed()) {
+            return $response->view('robots.allow', [], 200, ['Content-Type' => 'text/plain']);
+        }
+
+        return $response->view('robots.disallow', [], 200, ['Content-Type' => 'text/plain']);
```
Comment (rreynier) [inline src/app/Http/Controllers/RobotsTxtController.php]: Even though I am all about "returning" early.  Consider simplifying this into something like this: 

```php
$view = $indexingService->accessAllowed() ? 'robots.allow' : 'robots.disallow';

return $response->view($view, [], 200, ['Content-Type' => 'text/plain']);
```

File: src/app/Providers/IndexingAccessServiceProvider.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+     * Bootstrap the application services.
+     *
+     * @return void
+     */
+    public function boot()
+    {
+        $this->app->bind(IndexingAccessService::class, function($app) {
+           return new IndexingAccessService($app);
+        });
+    }
```
Comment (rreynier) [inline src/app/Providers/IndexingAccessServiceProvider.php]: I don't think this provider is actually needed. Everything here is auto-resolveable.

**Pull request merged**